editor.planx.uk/src/@planx/components/shared/Schema/InputFields/TextFieldInput.tsx

```diff
@@ -33,13 +36,22 @@ export const TextFieldInput: React.FC<Props<TextField>> = (props) => {
         required
         inputProps={{
           "aria-describedby": [
-            data.description ? DESCRIPTION_TEXT : "",
+            data.description
+              ? `${DESCRIPTION_TEXT} character-hint`
+              : "character-hint",
             errorMessage ? `${ERROR_MESSAGE}-${id}` : "",
           ]
             .filter(Boolean)
             .join(" "),
         }}
       />
+      {characterCountLimit && (
+        <CharacterCounter
+          limit={characterCountLimit}
+          count={fieldProps.value.length}
+          error={fieldProps.errorMessage ? true : false}
```

Comment:
This is to manage the style changes when an error is passed to the `<ErrorWrapper>` of the `<Input>` comp. Couldn't find a better way of managing this without there being strange styling.

Mostly comes when you go over the character limit, delete characters so you are within limit, but you don't submit. The alignment ends up being off between the Input error and the character counter error